azurerm/internal/services/machinelearning/machine_learning_compute_cluster_resource.go

```diff
@@ -0,0 +1,438 @@
+		Read:   resourceComputeClusterRead,
+		Update: resourceComputeClusterUpdate,
+		Delete: resourceComputeClusterDelete,
+
+		Importer: azSchema.ValidateResourceIDPriorToImport(func(id string) error {
+			_, err := parse.ComputeClusterID(id)
+			return err
+		}),
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(30 * time.Minute),
+			Read:   schema.DefaultTimeout(5 * time.Minute),
+			Update: schema.DefaultTimeout(30 * time.Minute),
+			Delete: schema.DefaultTimeout(30 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"machine_learning_workspace_id": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"location": azure.SchemaLocation(),
+
+			"vm_size": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"vm_priority": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"identity": {
+				Type:     schema.TypeList,
+				Required: true,
+				ForceNew: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"type": {
+							Type:     schema.TypeString,
+							Required: true,
+							ValidateFunc: validation.StringInSlice([]string{
+								string(machinelearningservices.SystemAssigned),
+							}, false),
+						},
+						"principal_id": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"tenant_id": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+					},
+				},
+			},
+
+			"scale_settings": {
+				Type:     schema.TypeList,
+				Required: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"max_node_count": {
+							Type:     schema.TypeInt,
+							Required: true,
+						},
+						"min_node_count": {
+							Type:     schema.TypeInt,
+							Required: true,
+						},
+						"node_idle_time_before_scale_down": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+					},
+				},
+			},
+
+			"description": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+
+			"subnet_resource_id": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+
+			"tags": tags.Schema(),
+		},
+	}
+}
+
+func resourceComputeClusterCreate(d *schema.ResourceData, meta interface{}) error {
+	mlWorkspacesClient := meta.(*clients.Client).MachineLearning.WorkspacesClient
+	mlComputeClient := meta.(*clients.Client).MachineLearning.MachineLearningComputeClient
+	ctx, cancel := timeouts.ForCreate(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	name := d.Get("name").(string)
+
+	// Get Machine Learning Workspace Name and Resource Group from ID
+	unparsedWorkspaceID := d.Get("machine_learning_workspace_id").(string)
+
+	workspaceID, err := parse.WorkspaceID(unparsedWorkspaceID)
```

Comment:
We could probably combine get and parse in here, like:
```suggestion
	// Get Machine Learning Workspace Name and Resource Group from ID
	workspaceID, err := parse.WorkspaceID(d.Get("machine_learning_workspace_id").(string))
```